common/txmgr/txmgr.go

```diff
@@ -487,16 +487,15 @@ func (b *Txm[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE, ADD, FEE_UN
 	if utils.IsZero(to) {
 		return etx, errors.New("cannot send native token to zero address")
 	}
-	etx = txmgrtypes.Tx[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE, ADD]{
+	newTx := txmgrtypes.NewTx[ADDR, TX_HASH]{
 		FromAddress:    from,
 		ToAddress:      to,
 		EncodedPayload: []byte{},
 		Value:          value,
 		FeeLimit:       gasLimit,
-		State:          TxUnstarted,
-		ChainID:        chainID,
+		Strategy:       NewSendEveryStrategy(),
 	}
-	err = b.txStore.InsertTx(&etx)
```

Comment:
That would be the ideal state, yes! 
Unfortunately we can't fully remove the `InsertTx` method yet.
The main reason is that multiple tests (example: [manually insert an unconfirmed Tx](https://github.com/smartcontractkit/chainlink/blob/1cf7f4ec0d327b10b0f98af836eda0e4228800d8/core/chains/evm/txmgr/broadcaster_test.go#L209-L210)) depend on the feature of inserting a Tx in a specific state (not UnStarted) and currently `CreateTransaction` doesn't allow that, it only creates transactions in UnStarted states. 
What I was thinking of doing is to move the   `InsertTx` method to the [TestEvmTxStore](https://github.com/smartcontractkit/chainlink/blob/45cb6ef1ccdf192df1c7e0a9ceadc881f1914162/core/chains/evm/txmgr/evm_tx_store.go#L43) interface that was recently created to allow its use only in tests.
I can do that in the next PR as there's still some work that needs to be done to enable that. [BCI-1397](https://smartcontract-it.atlassian.net/browse/BCI-1397)

Reply:
If we ensure that we can replace the coverage those tests offer that would be the optimal state.

I can create a ticket for replacing those tests and can look at it afterwards because I think it will require some dive deep.

Reply:
makes sense to me! and ideally we can eventually reduce the generic interface and move any test functions away :)

i'll be looking into more ways to reduce interfaces too

Reply:
I think we can remove from the tx_store.go file. After this PR, the only places that use InsertTx(), are unit-tests.
So we could extract out that method from this interface, and move to a test only interface.